relay-event-normalization/src/normalize/mod.rs

```diff
@@ -343,7 +343,23 @@ impl AiOperationTypeMap {
         if !self.is_enabled() {
             return None;
         }
-        self.operation_types.get(span_op).map(String::as_str)
+
+        // First try exact match by creating a Pattern from the span_op
+        let exact_key = Pattern::new(span_op).ok()?;
+        if self.operation_types.contains_key(&exact_key) {
+            return self.operation_types.get(&exact_key).map(String::as_str);
+        }
```

Comment:
I don't understand this part. Why would we set the type to the _key_ of the hash map instead of the value?

Reply:
The value of the hashmap is `String`, so if possible, we do only a hash lookup with `Pattern` value of `span_op`, if not we will iterate over all of the keys to search for a fallback. 

I don't really understand the question: "Why would we set the type to the key of the hash map instead of the value?"

If it helps and adds more context why it is done like this, we do the similar thing for model cost hash map: https://github.com/getsentry/relay/blob/38e6099f78e4e3b66952df7cf0151199523d4d7c/relay-event-normalization/src/normalize/mod.rs#L264

Reply:
We can do this without creating a `Pattern` for the key if you add a `Borrow` implementation to `Pattern`:

```rust
impl Borrow<str> for Pattern {
    fn borrow(&self) -> &str {
        &self.pattern
    }
}
```

From the [docs](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.get):

> The key may be any borrowed form of the map’s key type, but [Hash](https://doc.rust-lang.org/std/hash/trait.Hash.html) and [Eq](https://doc.rust-lang.org/std/cmp/trait.Eq.html) on the borrowed form must match those for the key type.

```suggestion
        if let Some(value) = self.operation_types.get(span_op) {
            return Some(value.as_str());
        }
```

Same for `cost_per_token`.

Reply:
Did it for `get_operation_type`, for `cost_per_token` i'll do it as a follow up because i need to change a bit more code (the reference will be returned instead the owned object), and it will have nicer commit associated with it